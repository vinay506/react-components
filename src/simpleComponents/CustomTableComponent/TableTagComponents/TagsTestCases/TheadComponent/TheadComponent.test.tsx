import React from 'react';
import { shallow } from 'enzyme';
import { CustomThead, CustomTableRow, CustomThCell,CustomTdCell } from '../../index';
import renderer from 'react-test-renderer'

const headerColumns = [{ title: "Tests", styles: { fontWeight: 'bold', height: '45px' } }];
const getCustomTableHeaderCells = () => {
    return (headerColumns || []).map((column: any, index: number) => {
        return <CustomThCell style={column.styles} key={'hdcol-' + index}>{column.title}</CustomThCell>
    })
}
describe('<Custom Theader Tag >', function () {
    let props: any;
    let tree: any;
    let wrapper: any;
    beforeEach(() => {
        props = {};
        wrapper = shallow(<CustomThead {...props}> <CustomTableRow headerRow>
            {getCustomTableHeaderCells()}
        </CustomTableRow></CustomThead>);
    });

    // Snapshot test
    it('should render', () => {
        tree = renderer.create(<CustomThead {...props}> <CustomTableRow headerRow>
            {getCustomTableHeaderCells()}
        </CustomTableRow></CustomThead>).toJSON()
        expect(tree).toMatchSnapshot()
    })

    it('checks with the className of Table header', () => {
        const props = wrapper.find('thead').props();
        expect(props.className).toEqual('rmwc-data-table__head');
    })

    it('checks the flag in custom table row', () => {
        const customTableRow = wrapper.find(CustomTableRow);
        const props = customTableRow.props();
        expect(props.headerRow).toBeTruthy()
    })

    it('checks the existing of CustomThCell cells in tr', () => {
        const customTableRow = wrapper.find(CustomTableRow);
        const thcells = customTableRow.find(CustomThCell);
        expect(thcells).toHaveLength(1);
    })

    it('checks to make sure that it holds CustomThCell cells only in tr', () => {
        const customTableRow = wrapper.find(CustomTableRow);
        const tdcells = customTableRow.find(CustomTdCell);
        expect(tdcells).toHaveLength(0);
    })


})