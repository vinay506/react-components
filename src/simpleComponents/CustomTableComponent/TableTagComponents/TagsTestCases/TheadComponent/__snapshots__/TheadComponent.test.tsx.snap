// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Custom Theader Tag > should render 1`] = `
<thead
  className="rmwc-data-table__head"
>
   
  <tr
    className="rmwc-data-table__row mdc-data-table__header-row"
  >
    <th
      className="rmwc-data-table__cell mdc-data-table__header-cell"
      style={
        Object {
          "fontWeight": "bold",
          "height": "45px",
        }
      }
    >
      Tests
    </th>
  </tr>
</thead>
`;
